operate/services/manage.py

```diff
@@ -323,53 +363,92 @@ def deploy_service_onchain(  # pylint: disable=too-many-statements
+        """
+        Deploy as service on-chain
+
+        :param hash: Service hash
+        """
+
+        self.logger.info(f"_deploy_service_onchain_from_safe {chain_id=}")
+        service = self.load_or_create(hash=hash)
+        chain_config = service.chain_configs[chain_id]
+        ledger_config = chain_config.ledger_config
+        chain_data = chain_config.chain_data
+        user_params = chain_config.chain_data.user_params
         keys = service.keys
         instances = [key.address for key in keys]
-        wallet = self.wallet_manager.load(service.ledger_config.type)
-        sftxb = self.get_eth_safe_tx_builder(service=service)
-        if user_params.use_staking and not sftxb.staking_slots_available(
-            staking_contract=STAKING[service.ledger_config.chain]
-        ):
-            raise ValueError("No staking slots available")
+        wallet = self.wallet_manager.load(ledger_config.type)
+        sftxb = self.get_eth_safe_tx_builder(ledger_config=ledger_config)
 
-        if service.chain_data.token > -1:
+        # TODO fix this
+        os.environ["CUSTOM_CHAIN_RPC"] = ledger_config.rpc
+        os.environ["OPEN_AUTONOMY_SUBGRAPH_URL"] = "https://subgraph.autonolas.tech/subgraphs/name/autonolas-staging"
+
+        current_agent_id = None
+        if chain_data.token > -1:
             self.logger.info("Syncing service state")
-            info = sftxb.info(token_id=service.chain_data.token)
-            service.chain_data.on_chain_state = OnChainState(info["service_state"])
-            service.chain_data.instances = info["instances"]
-            service.chain_data.multisig = info["multisig"]
+            info = sftxb.info(token_id=chain_data.token)
+            chain_data.on_chain_state = OnChainState(info["service_state"])
+            chain_data.instances = info["instances"]
+            chain_data.multisig = info["multisig"]
+            current_agent_id = info["canonical_agents"][0]  # TODO Allow multiple agents
             service.store()
-        self.logger.info(f"Service state: {service.chain_data.on_chain_state.name}")
+        self.logger.info(f"Service state: {chain_data.on_chain_state.name}")
+
+        if user_params.use_staking:
+            staking_params = sftxb.get_staking_params(
+                staking_contract=STAKING[ledger_config.chain][user_params.staking_program_id],
+            )
+        else:  # TODO fix this - using pearl beta params
+            staking_params = dict(
+                agent_ids=[25],
+                service_registry="0x9338b5153AE39BB89f50468E608eD9d764B755fD",  # nosec
+                staking_token="0xcE11e14225575945b8E6Dc0D4F2dD4C570f79d9f",  # nosec
+                service_registry_token_utility="0xa45E64d13A30a51b91ae0eb182e88a40e9b18eD8",  # nosec
+                min_staking_deposit=20000000000000000000,
+                activity_checker="0x155547857680A6D51bebC5603397488988DEb1c8"  # nosec
+            )
 
         if user_params.use_staking:
```

Comment:
if user_params.use_staking: 
else:
## using pearl beta params??

Reply:
This is a hotfix to avoid having uninitialized variables. If use_staking = false, it will by default initialize with (possible unused) Pearl beta variables.